### ISSUE TITLE
[Host] Abandon/dead letter message

### ISSUE THREAD

**Issue by EtherZa** [zarusz/SlimMessageBus#356]
SMB does not provide an out of the box way to dead letter messages for the transports that support it. It is currently possible for RabbitMQ through a nack in a `IConsumerErrorHandler`, but not for Azure Service Bus.

ASB messages that throw an exception will be retried n times before finally being sent to the DLQ. 

If an exception can be determined to be non-transient (and therefore will not be resolved in a subsequent execution), it would be beneficial to send the message to the DLQ immediately.

**Comment by zarusz** [comment]
Yes, we could add this as an asb transport specific feature. Perhaps even unify across all transports (but not sure about if it's worth unifying).

**Comment by zarusz** [comment]
What would be good is also gave the ability to set application properties (headers) from the error handler or consumer when message is abandoned on the asb transports. This helps to convey partially completed work and have it available for the next message retry.

**Comment by zarusz** [comment]
Closed via #357.

Also see another feature request: #358
